From eb18d2edc2463dd12498e54f745068e4fdbf2b08 Mon Sep 17 00:00:00 2001
From: Kha Tran <kha.tran.px@rvc.renesas.com>
Date: Tue, 25 Sep 2018 18:10:36 +0700
Subject: [PATCH 420/442] usb: gadget: composite: Avoid working with NULL
 pointer of cdev

This patch checks and returns if cdev is NULL.

Signed-off-by: Kha Tran <kha.tran.px@rvc.renesas.com>
Signed-off-by: Hao Bui <hao.bui.yg@renesas.com>
---
 drivers/usb/gadget/composite.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/usb/gadget/composite.c b/drivers/usb/gadget/composite.c
index bb72d55..9c33196 100644
--- a/drivers/usb/gadget/composite.c
+++ b/drivers/usb/gadget/composite.c
@@ -1896,6 +1896,10 @@ void composite_disconnect(struct usb_gadget *gadget)
 	struct usb_composite_dev	*cdev = get_gadget_data(gadget);
 	unsigned long			flags;
 
+	/*workaround: check if cdev is NULL. Exit function*/
+	if (!cdev)
+		return;
+
 	/* REVISIT:  should we have config and device level
 	 * disconnect callbacks?
 	 */
-- 
2.7.4

